src/components/pages/Dashboard.jsx

```diff
@@ -0,0 +1,164 @@
+
+import DayJS from 'dayjs';
+import React, { Component } from 'react';
+
+import Loading from '../elements/Loading';
+
+import DashboardUtils from '../../utils/dashboard';
+
+
+
+import '../../css/elements/dashboard.css';
+import '../../css/elements/filter.css';
+import '../../css/pages/dayoff.css';
+import Language from '../../utils/language';
+import DashboardActionToTake from '../elements/dashboard/dashboardActionToTake';
+import DashboardActivityFeed from '../elements/dashboard/dashboardLastActions';
+import DashboardMetrics from '../elements/dashboard/dashboardMetrics';
+
+class Dashboard extends Component {
+    constructor(props) {
+        super(props);
+        const filter = {
+            start: DayJS().format('YYYY-MM-DD'),
+            end: DayJS().format('YYYY-MM-DD'),
+            status: 'confirmed'
+        };
+        const filterActions = {
+            start: DayJS().format('YYYY-MM-DD'),
+            status: 'pending'
+        }
+
+        this.state = {
+            loading: true,
+            filter,
+            filterActions,
+            numberDaysoff: 0,
+            numberSlackUsers: 0,
+            numberActions: 0,
+            activity: null,
+            formDialog: false,
+            dayoffId: null
+        };
+    }
+
+    async componentDidMount() {
+    // chargement données page
+        await DashboardUtils.init(this);
+    }
+
+    // eslint-disable-next-line react/no-unused-class-component-methods
+    handleInit = (data) => {
+        this.setState({
+            loading: false,
+            ...data
+        });
+    };
+
+    updateState = (newState) => {
```

Comment:
A quoi sert cette fonction ? Pourquoi ne pas utiliser this.setState directement ?